CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKHitTester.java

```diff
@@ -52,17 +110,158 @@ public CVKHitTester(CVKVisualProcessor parent) {
     }
     
     @Override
-    public int Initialise(CVKDevice cvkDevice) { return VK_SUCCESS;}  
+    public int Initialise(CVKDevice cvkDevice) { 
+        this.cvkDevice = cvkDevice;  
+        return VK_SUCCESS;
+    }
     
     @Override
-    public void Destroy() {}
+    public void Destroy() {
+        DestroyFrameBuffer();
+        DestroyImage();
+        DestroyCommandBuffer();
+        
+        CVKAssert(vkFrameBufferHandle == null);
+        CVKAssert(cvkImage == null);
+        CVKAssert(commandBuffer == null);
+    }
     
     
     // ========================> Swap chain <======================== \\    
     
     @Override
     protected int DestroySwapChainResources() { return VK_SUCCESS; }
     
+    private int CreateSwapChainResources() {
+        cvkVisualProcessor.VerifyInRenderThread();
+        CVKAssert(cvkSwapChain != null);
+        int ret = VK_SUCCESS;
+
+                
+        // We only need to recreate these resources if the number of images in 
+        // the swapchain changes or if this is the first call after the initial
+        // swapchain is created.
+        if (swapChainImageCountChanged) {
+            ret = CreateImages();
+            if (VkFailed(ret)) { return ret; }            
+
+            ret = CreateFrameBuffer();
+            if (VkFailed(ret)) { return ret; }
+
+             ret = CreateCommandBuffer();
+            if (VkFailed(ret)) { return ret; }
+            
+        } else {
+            // This is the resize path, image count is unchanged.
+        }
+        
+        swapChainResourcesDirty = false;
+        swapChainImageCountChanged = false;
+        
+        return ret;
+    } 
+    
+    
+    // ========================> Image <======================== \\
+    
+    private int CreateImages() {
+        CVKAssert(cvkSwapChain != null);
+        CVKAssert(cvkSwapChain.GetDepthFormat() != VK_FORMAT_UNDEFINED);
+        
+        int ret = VK_SUCCESS;
+   
+        try (MemoryStack stack = stackPush()) {
+            
+            int textureWidth = cvkSwapChain.GetWidth();
+            int textureHeight = cvkSwapChain.GetHeight();
+            int requiredLayers = 1;
+                        
+            // Create destination color image            
+            cvkImage = CVKImage.Create(cvkDevice, 
+                                            textureWidth, 
+                                            textureHeight, 
+                                            requiredLayers, 
+                                            //colorFormat, // Format TODO Not sure what the format should be - look at GL version
```

Comment:
A new format, R8 would be ideal, but would need new shaders.  Maybe we could use the RGBA channels in a quarter size image and then just use the same shaders?

Reply:
Yeah I know, I'm still working on this part, also not sure a format like that exists. Will get it working first then looking at reducing size.